cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/Lookup.java

```diff
@@ -16,15 +16,43 @@
 
 package co.cask.cdap.etl.api;
 
+import co.cask.cdap.api.data.schema.Schema;
+
+import java.util.Collection;
 import java.util.Map;
 import java.util.Set;
+import javax.annotation.Nullable;
 
 /**
  * Exposes read-only lookup operations on datasets.
  *
- * @param <T> the type of object that will be returned for a lookup
+ * @param <T> the type of object that will be returned for a lookup with String as the key
+ * @param <R> the raw type of object that will be returned for lookups with byte[] as the key
```

Comment:
The comment wasn't about returning a StructuredRecord, but was about why we need to add this second parameter. The value is already a generic, shouldn't there be a way to get a `Lookup<byte[]>` instead of a `Lookup<String>`? Maybe based on the arguments passed into provide()?

Otherwise it's really strange, I have `Lookup<String, byte[]>` then `lookup("abc")` returns a different type than `lookup(Bytes.toBytes("abc"))`. If anything, I would expect the second parameter to be for the key type.